m4/check_java_support.m4

```diff
@@ -4,12 +4,17 @@ AU_ALIAS([AC_CHECK_GRADLE_VERSION], [AX_CHECK_GRADLE_VERSION])
 AC_DEFUN([AX_READLINK],
 [
   READLINK_TARGET=[$1]
-  cd $(dirname "$READLINK_TARGET")
-  while test -L "$READLINK_TARGET"; do
-    READLINK_TARGET=$(readlink "$READLINK_TARGET")
+
+  if test -n "$READLINK_TARGET"; then
     cd $(dirname "$READLINK_TARGET")
-  done
-  echo $(pwd -P)/$(basename "$READLINK_TARGET")
+    while test -L "$READLINK_TARGET"; do
+      READLINK_TARGET=$(readlink "$READLINK_TARGET")
+      cd $(dirname "$READLINK_TARGET")
+    done
+    echo $(pwd -P)/$(basename "$READLINK_TARGET")
+  else
+    echo "$READLINK_TARGET"
```

Comment:
test -n means string is nonzero length, and we are now in the else branch, so now READLINK_TARGET is empty. Why are we echo-ing it? What does this patch fix?

Reply:
It prevents returning an invalid path when the target is empty.
(`dirname` returns the current working dir when its first parameter is empty)

Reply:
As we discussed irl, the else brach is not necessary. It does not cause problem either, but not necessary.

Reply:
Removed, thanks.